app/jobs/verify_fixity_job.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class VerifyFixityJob < ApplicationJob
+  queue_as Atc::Queues::VERIFY_FIXITY
+
+  def perform(stored_object_id)
+    stored_object = StoredObject.find(stored_object_id)
+
+    # For now, handle only AWS fixity check
+    return unless stored_object.storage_provider.aws?
+
+    existing_fixity_verification_record = FixityVerification.find_by(stored_object: stored_object)
+    return if existing_fixity_verification_record&.pending?
+
+    if existing_fixity_verification_record
+      process_existing_fixity_verification_record existing_fixity_verification_record
+    end
+    fixity_verification_record = create_pending_fixity_verification stored_object
+    provider_fixity_check = instantiate_provider_fixity_check fixity_verification_record
+    verify_fixity(fixity_verification_record, provider_fixity_check)
+  end
+
+  def process_existing_fixity_verification_record(existing_fixity_verification_record)
+    return if existing_fixity_verification_record.pending?
+
+    # for now, just delete the existing FixityVerification. In later implementation, copy the info
+    # into an entry in the new table (PastFixityVerifications).
+    existing_fixity_verification_record.destroy
+  end
+
+  def create_pending_fixity_verification(stored_object)
+    fixity_verification_record = FixityVerification.create!(source_object: stored_object.source_object,
+                                                            stored_object: stored_object)
+    fixity_verification_record.pending! # saves to the database
```

Comment:
Suggestion for later (not critical right now), but you can set the status to pending in the `create!` call to avoid a second save operation on this line.

Reply:
Actually, I just double checked the schema and the default value of status is `pending`, so you could remove the line above and I would expect things to work the same way.

Reply:
I like adding the pending status to the create! call (makes it explicit). I'll add the change in a later PR.

Reply:
Amending my previous comment: I updated the code in this PR to set the status to pending in the `create!` call.